File: langs/add.js
function sumOfDigits(n) {
    let total = 0;
    while (n > 0) {
        total += n % 10;
        n = Math.floor(n / 10);
    }
    return total;
}

const readline = require("readline").createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question("Enter a number: ", (num) => {
    console.log("Sum of digits:", sumOfDigits(parseInt(num)));
    readline.close();
});

File: langs/add.cpp
#include <iostream>
using namespace std;

int sum_of_digits(int n) {
    int total = 0;
    while (n > 0) {
        total += n % 10;
        n /= 10;
    }
    return total;
}

int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;
    cout << "Sum of digits: " << sum_of_digits(num) << endl;
    return 0;
}
